src/services/actions.js

```diff
@@ -0,0 +1,183 @@
+				result: createAction,
+			})
+		} catch (error) {
+			if (error instanceof UniqueConstraintError) {
+				return responses.failureResponse({
+					message: 'ACTION_ALREADY_EXISTS',
+					statusCode: httpStatusCode.bad_request,
+					responseCode: 'CLIENT_ERROR',
+				})
+			} else {
+				return responses.failureResponse({
+					message: error.message || error,
+					statusCode: httpStatusCode.bad_request,
+					responseCode: 'CLIENT_ERROR',
+				})
+			}
+		}
+	}
+
+	/**
+	 * Update Action
+	 * @method
+	 * @name update
+	 * @param {Integer} id - Action Id
+	 * @param {String} bodyData - action data to be updated
+	 * @returns {JSON} - action update response
+	 */
+	static async update(id, bodyData) {
+		try {
+			const [updateCount, updatedAction] = await actionQueries.updateOne({ id: id }, bodyData, {
+				returning: true,
+				raw: true,
+			})
+
+			if (updateCount === 0) {
+				throw new Error('ACTION_NOT_FOUND')
+			}
+
+			return responses.successResponse({
+				statusCode: httpStatusCode.ok,
+				message: 'ACTION_UPDATED',
+				result: updatedAction[0],
+			})
+		} catch (error) {
+			if (error instanceof UniqueConstraintError) {
+				return responses.failureResponse({
+					message: 'ACTION_ALREADY_EXISTS',
+					statusCode: httpStatusCode.bad_request,
+					responseCode: 'CLIENT_ERROR',
+				})
+			} else {
+				return responses.failureResponse({
+					message: error.message || error,
+					statusCode: httpStatusCode.bad_request,
+					responseCode: 'CLIENT_ERROR',
+				})
+			}
+		}
+	}
+
+	/**
+	 * list actions.
+	 * @method
+	 * @name list
+	 * @param {Integer} page - Page number
+	 * @param {Integer} page - Page size limit
+	 * @param {String} search - Search text.
+	 * @returns {JSON} - actions list response.
+	 */
+
+	static async list(page, limit, search) {
+		try {
+			let result = {
+				data: [],
+				count: 0,
+			}
+			const offset = common.getPaginationOffset(page, limit)
+
+			const filter = {
+				code: { [Op.iLike]: `%${search}%` },
+			}
+			const options = {
+				offset,
+				limit,
+			}
+			const attributes = ['id', 'code', 'status']
+			const actions = await actionQueries.findAllActions(filter, attributes, options)
+
+			if (actions.count <= 0) {
+				return responses.successResponse({
+					statusCode: httpStatusCode.ok,
+					message: 'ACTIONS_FETCHED',
+					result: result,
+				})
+			}
+
+			result.data = actions.rows
+			result.count = actions.count
+
+			return responses.successResponse({
+				statusCode: httpStatusCode.ok,
+				message: 'ACTIONS_FETCHED',
+				result: result,
+			})
+		} catch (error) {
+			throw error
+		}
+	}
+
+	/**
+	 * Delete action
+	 * @method
+	 * @name delete
+	 * @param {String} id - action id.
+	 * @returns {JSON} - action deleted response.
+	 */
+
+	static async delete(id) {
+		try {
+			const action = await actionQueries.findById(id)
+			if (!action) {
+				throw new Error('ACTION_NOT_FOUND')
+			}
+
+			const activities = await activitiesQueries.findAllActivities(
```

Comment:
@priyanka-TL won't this query be non-optimal? Without DC and index?

Reply:
@aks30 we don't have another option here. When deleting the action, we can't include the org ID. We'll need to verify if any other organisation has activities linked to the action being deleted. Its deletion so its not a frequent activity.